backends/platform/3ds/osystem-graphics.cpp

```diff
@@ -30,32 +30,72 @@
-	 GX_TRANSFER_SCALING(GX_TRANSFER_SCALE_NO))
+#define DISPLAY_TRANSFER_FLAGS(in, out)                         \
+        (GX_TRANSFER_FLIP_VERT(0) | GX_TRANSFER_OUT_TILED(0) |  \
+         GX_TRANSFER_RAW_COPY(0) | GX_TRANSFER_IN_FORMAT(in) |  \
+         GX_TRANSFER_OUT_FORMAT(out) |                          \
+         GX_TRANSFER_SCALING(GX_TRANSFER_SCALE_NO))
+#define TEXTURE_TRANSFER_FLAGS(fmt)                             \
+        (GX_TRANSFER_FLIP_VERT(1) | GX_TRANSFER_OUT_TILED(1) |  \
+         GX_TRANSFER_RAW_COPY(0) | GX_TRANSFER_IN_FORMAT(fmt) | \
+         GX_TRANSFER_OUT_FORMAT(fmt) |                          \
+         GX_TRANSFER_SCALING(GX_TRANSFER_SCALE_NO))
 
 namespace _3DS {
 
-void OSystem_3DS::initGraphics() {
+/* Group the various enums, values, etc. needed for
+ * each graphics mode into instaces of GfxMode3DS */
+GfxMode3DS _modeRGBA8 = { Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0),
+                          Graphics::PixelFormat(3, 8, 8, 8, 0, 0, 8, 16, 0),
+                          GSP_BGR8_OES, GPU_RB_RGBA8, GPU_RGBA8,
+                          DISPLAY_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGBA8, GX_TRANSFER_FMT_RGB8),
+                          TEXTURE_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGBA8) };
+GfxMode3DS _modeRGB565 = { Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0),
+                           Graphics::PixelFormat(2, 5, 6, 5, 0, 11, 5, 0, 0),
+                           GSP_RGB565_OES, GPU_RB_RGB565, GPU_RGB565,
+                           DISPLAY_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB565, GX_TRANSFER_FMT_RGB565),
+                           TEXTURE_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB565) };
+GfxMode3DS _modeRGB8 = { Graphics::PixelFormat(3, 0, 0, 0, 8, 0, 8, 16, 0),
+                         Graphics::PixelFormat(3, 0, 0, 0, 8, 0, 8, 16, 0),
+                         GSP_BGR8_OES, GPU_RB_RGB8, GPU_RGB8,
+                         DISPLAY_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB8, GX_TRANSFER_FMT_RGB8),
+                         TEXTURE_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB8) };
+GfxMode3DS _modeRGB555 = { Graphics::PixelFormat(2, 5, 5, 5, 0, 10, 5, 0, 0),
+                           Graphics::PixelFormat(2, 5, 5, 5, 1, 11, 6, 1, 0),
+                           GSP_RGB5_A1_OES, GPU_RB_RGBA5551, GPU_RGBA5551,
+                           DISPLAY_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB5A1, GX_TRANSFER_FMT_RGB5A1),
+                           TEXTURE_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB5A1) };
+GfxMode3DS _modeRGB5A1 = { Graphics::PixelFormat(2, 5, 5, 5, 1, 11, 6, 1, 0),
+                           Graphics::PixelFormat(2, 5, 5, 5, 1, 11, 6, 1, 0),
+                           GSP_RGB5_A1_OES, GPU_RB_RGBA5551, GPU_RGBA5551,
+                           DISPLAY_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB5A1, GX_TRANSFER_FMT_RGB5A1),
+                           TEXTURE_TRANSFER_FLAGS(GX_TRANSFER_FMT_RGB5A1) };
+GfxMode3DS _modeCLUT8 = _modeRGBA8;
+
+GfxMode3DS *gfxModes[6] = { &_modeRGBA8, &_modeRGB565, &_modeRGB8,
```

Comment:
There is no need to set the size of the array here. The array can be `static const`.